code/modules/mob/living/simple_animal/hostile/retaliate/red_dragon.dm

```diff
@@ -32,6 +32,46 @@
 	deathmessage = "collapses to the floor with a final roar, the impact rocking the ground."
 	footstep_type = FOOTSTEP_MOB_HEAVY
 
+/mob/living/simple_animal/hostile/retaliate/voiddragon/red/void_pull(atom/target)
+	var/turf/T = get_turf(src)
+	playsound(T, 'sound/magic/charging_lightning.ogg', 100, TRUE)
+
+	// Visual effect
+	new /obj/effect/temp_visual/dragon_swirl(T)
+
+	// Pull in all mobs within range
+	for(var/mob/living/L in range(7, src))
+		if(L == src || L.stat == DEAD)
+			continue
+		L.visible_message(span_warning("[L] is pulled toward [src]!"))
+
+		var/throw_dist = get_dist(L, src)
+
+		// Calculate throw speed based on distance
+		var/throw_speed = max(1, 3 - round(throw_dist / 3))
+		L.throw_at(src, throw_dist, throw_speed)
+		L.apply_damage(10, BRUTE)
+
+	addtimer(CALLBACK(src, PROC_REF(void_pull_aftermath)), 1 SECONDS)
+
+
+/mob/living/simple_animal/hostile/retaliate/voiddragon/red/void_explosion_detonate(turf/target)
+	if(!target)
+		return
+
+	playsound(target, 'sound/magic/disintegrate.ogg', 200, TRUE)
+
+	new /obj/effect/temp_visual/dragon_explosion(target)
+
+	for(var/mob/living/L in range(3, target))
+		var/dist = get_dist(target, L)
+		var/damage = 30 * (1 - (dist / 4)) // 30 damage at epicenter, scaling down with distance
+		L.apply_damage(damage, BRUTE)
+
+		var/throw_dir = get_dir(target, L)
+		L.throw_at(get_edge_target_turf(L, throw_dir), 3, 2)
+
+
```

Comment:
Instead of remaking procs to remove one line add a variable to do the change you want. 
This was the same with the inhumen healing miracle.

Reply:
okay